docs/resources/service_account_token.md

```diff
@@ -0,0 +1,83 @@
+---
+page_title: "rancher2_service_account_token Resource"
+---
+
+# rancher2\_service\_account\_token Resource
+
+Provides a Rancher v2 Token resource, specifically to create service account tokens. Service accounts tokens are tokens for other users (service accounts) than the Rancher v2 provider user. To create a service account token the username/password for the Rancher User must be known.
+
+There are 2 kind of tokens:
+- no scoped: valid for global system.
+- scoped: valid for just a specific cluster (`cluster_id` should be provided).
+
+Tokens can only be created for a Rancher User with at least the `user-base` global role binding in order to enable user login.
+
+Tokens can't be updated once created. Any diff in token data will recreate the token. If any token expire, Rancher2 provider will generate a diff to regenerate it.
+
+## Example Usage
+
+```hcl
+# Create a rancher2 Token
+resource "rancher2_user" "foo" {
+  name = "foo"
+  username = "foo"
+  password = "changeme"
+  enabled = true
+}
+
+resource "rancher2_global_role_binding" "foo-login" {
+  name = "foo-login-binding"
+  global_role_id = "user-base"
+  user_id = rancher2_user.foo.id
+}
+
+resource "rancher2_service_account_token" "foo" {
+  username = rancher2_user.foo.username
+  password = rancher2_user.foo.password
+  description = "foo token"
+  ttl = 0
+
+  depends_on = [
+    rancher2_global_role_binding.foo-login
+  ]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `username` - (Required/ForceNew) The user username (string)
```

Comment:
Yes, tokens can't be updated, any change to the resource basically becomes a recreate operation. See the existing rancher2_token resource for reference, it also uses the `ForceNew` behaviour.
A change to the `username` field means a token is requested for a completely different user(account). Hence I  marked them as `ForceNew` as well.

Reply:
Thank you for the docs link in the other comment. This makes sense to me, since `username` and `password` are part of the token_resource here then they should also be `ForceNew` because they will be included in the recreate operation.